src/megameklab/com/ui/supportvehicle/SVMainUI.java

```diff
@@ -43,27 +42,13 @@
     private SVBuildTab buildTab;
     private FluffTab fluffTab;
     private SVStatusBar statusbar;
-    private JPanel masterPanel = new JPanel();
 
     public SVMainUI() {
 
         super();
         createNewUnit(Entity.ETYPE_SUPPORT_TANK, false, false);
         setTitle(getEntity().getChassis() + " " + getEntity().getModel() + ".blk");
-        MenuBarCreator menubarcreator = new MenuBarCreator(this);
-        setJMenuBar(menubarcreator);
-        JScrollPane scroll = new JScrollPane();
-        scroll.setHorizontalScrollBarPolicy(ScrollPaneConstants.HORIZONTAL_SCROLLBAR_AS_NEEDED);
-        scroll.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED);
-        scroll.getVerticalScrollBar().setUnitIncrement(20);
-        scroll.setViewportView(masterPanel);
-        scroll.setBorder(BorderFactory.createEmptyBorder());
-        this.add(scroll);
-
-        reloadTabs();
-        setVisible(true);
-        repaint();
-        refreshAll();
+        finishSetup();
         MechSummaryCache.getInstance();
```

Comment:
@neoancient Some of the MainUI() methods had this line at the end, even though this is actually called at the very top of the constructor for MegaMekLabMainUI from which all of these inherit. Is it redundant or is something else going on here?

Reply:
Actually having had time to thing about it I think the idea is to start the cache loading thread to cut or eliminate the wait the first time a unit is loaded from the cache. It would be better placed somewhere in the application startup process if it isn't already.

Reply:
it is already done in the super() constructor.